rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -348,9 +350,17 @@ private long setCorrectEndBlock(long endBlock, long fileLength, String remotePat
 
   private static synchronized void initializeCache(final Configuration conf)
   {
+    try {
+      CacheUtil.createCacheDirectories(conf);
+    }
+    catch (FileNotFoundException e) {
+      log.error("Cache directories could not be created", e);
+      System.exit(0);
```

Comment:
We should not be doing System.exit here. Let the caller decide what needs to be done. BookKeeperServer initializes BookKeeper class. We should catch the exception there and do the needful.